modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -110,3 +110,38 @@ def func_to_apply(partition, row_internal_indices, col_internal_indices, item):
     md_df._query_compiler._modin_frame = new_modin_frame
 
     df_equals(md_df, pd_df)
+
+
+@pytest.mark.skipif(
+    Engine.get() != "Dask" and Engine.get() != "Ray",
+    reason="Rebalancing partitions is only supported for Dask and Ray engines",
+)
+def test_rebalance_partitions():
+    small_dfs = [
+        pd.DataFrame(
+            [[i + j for j in range(0, 100)]],
+            columns=[f"col{j}" for j in range(1, 101)],
+            index=pd.Index([i - 1]),
+        )
+        for i in range(1, 10001, 100)
+    ]
+    large_df = pd.concat(small_dfs)
```

Comment:
Could you also add a case that forces us to split a partition? e.g. concat a DF with 10_000 rows to 5  dfs with 1 row each? e.g. this produces splitting with `NPartitions` of 16:

```
pd.concat([pd.DataFrame([[i] for i in range(1000)])] + [pd.DataFrame([[i]]) for i in range(16)])
```

you need enough partitions in the first partition for the partitioning to be imbalanced enough to trigger repartitioning.

Reply:
That makes a lot of sense - will go ahead and add!